packages/db-objects-file/lib/objects/objectsInMemServerRedis.js

```diff
@@ -757,24 +723,29 @@ class ObjectsInMemoryServer extends ObjectsInMemoryFileDB {
                             notExists: true
                         }];
                     }
-                    const response = [];
-                    const baseName = name + ((!name.length || name.endsWith('/')) ? '' : '/');
-                    res.forEach(arr => {
-                        let entryId = id;
-                        if (arr.isDir) {
-                            if (entryId === '' || entryId === '*') {
-                                entryId = arr.file;
-                                arr.file = '_data.json'; // We return a "virtual file" to mark the directory as existing
-                            } else {
-                                arr.file += '/_data.json'; // We return a "virtual file" to mark the directory as existing
-                            }
+                } catch (err) {
+                    if (!err.toString().endsWith(utils.ERRORS.ERROR_NOT_FOUND)) {
+                        return void handler.sendError(responseId, new Error(`ERROR readDir id=${id}: ${err.message}`));
+                    }
+                    res = [];
+                }
+                const response = [];
+                const baseName = (name || '') + ((!name || !name.length || name.endsWith('/')) ? '' : '/');
```

Comment:
_normalizeId defaults `name` to `''`, so these checks shouldn't be necessary. For readability I would change this to
```js
let baseName = name;
if (!baseName.endsWith('/')) baseName += '/';
```

Reply:
Yes I added that before I decided to fix it in the methid itself:-) I Change it